### PR TITLE
Raise error when Pydantic dataclasses are used on models

### PR DESCRIPTION
**Pull request by Viicos**
Fixes https://github.com/pydantic/pydantic/issues/9383

### CODE REVIEW

File: pydantic/dataclasses.py [hunk 1]
```diff
@@ -190,6 +192,12 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         Returns:
             A Pydantic dataclass.
         """
+        if is_model_class(cls):
+            raise PydanticUserError(
+                f'Cannot create a Pydantic dataclass from {cls.__name__} as it is already a Pydantic model',
+                code=None,
```
Comment (Viicos) [inline pydantic/dataclasses.py:198]: Wondering if a new code should be created

Comment (hramezani) [reply pydantic/dataclasses.py:198]: I think we need a new code

**Comment by Viicos** [comment]
Waiting for directions regarding the `test_public_internal` test failure

File: pydantic/dataclasses.py [hunk 2]
```diff
@@ -190,6 +192,12 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         Returns:
             A Pydantic dataclass.
         """
+        if is_model_class(cls):
```
Comment (sydney-runkle) [inline pydantic/dataclasses.py:195]: I'd make the same suggestion re the test failures, let's see if this approach fixes things: https://github.com/pydantic/pydantic/pull/9398#discussion_r1600373980

**Pull request merged**